ironfish/src/blockchain/blockchain.test.ts

```diff
@@ -759,74 +750,44 @@ describe('Blockchain', () => {
     // Now create the double spend
     await node.wallet.updateHead()
     const tx = await useTxFixture(node.wallet, accountA, accountB)
-    const doubleSpend = tx.getSpend(0)
-
-    const treeSize = await node.chain.nullifiers.size()
-
-    // The nullifier is not found in the tree
-    await expect(node.chain.nullifiers.contains(doubleSpend.nullifier)).resolves.toBe(false)
-    await expect(
-      node.chain.nullifiers.contained(doubleSpend.nullifier, treeSize),
-    ).resolves.toBe(false)
 
-    // Let's spend the transaaction for the first time
+    // Spend the transaaction for the first time
     const block3 = await useMinerBlockFixture(node.chain, 3, undefined, undefined, [tx])
     await expect(node.chain).toAddBlock(block3)
 
-    // The nullifier is not found at the old tree size, but is found at the new tree size
-    await expect(node.chain.nullifiers.contains(doubleSpend.nullifier)).resolves.toBe(true)
-    await expect(
-      node.chain.nullifiers.contained(doubleSpend.nullifier, treeSize),
-    ).resolves.toBe(false)
-    await expect(
-      node.chain.nullifiers.contained(doubleSpend.nullifier, treeSize + tx.spendsLength()),
-    ).resolves.toBe(true)
-
-    // Let's spend the transaction a second time
+    // Spend the transaction a second time
     const block4 = await useMinerBlockFixture(node.chain, 4, undefined, undefined, [tx])
-    await expect(node.chain).toAddBlock(block4)
-
-    // We've now added a double spend, we can see that because of a bug in the
-    // MerkleTree implementation the nullifier is no longer found at the tree
-    // size before this block, but moved to the tree size at block3
-    await expect(node.chain.nullifiers.contains(doubleSpend.nullifier)).resolves.toBe(true)
-    await expect(
-      node.chain.nullifiers.contained(doubleSpend.nullifier, treeSize + tx.spendsLength()),
-    ).resolves.toBe(false)
-    await expect(
-      node.chain.nullifiers.contained(
-        doubleSpend.nullifier,
-        treeSize + tx.spendsLength() + tx.spendsLength(),
-      ),
-    ).resolves.toBe(true)
-
-    // Now we set our fix to activate at sequence 5 so this block will not let
-    // us add the transaction a third time
-    const block5 = await useMinerBlockFixture(node.chain, 5, undefined, undefined, [tx])
-    await expect(node.chain.addBlock(block5)).resolves.toMatchObject({
+    await expect(node.chain.addBlock(block4)).resolves.toMatchObject({
       isAdded: false,
       reason: VerificationResultReason.DOUBLE_SPEND,
     })
   })
 
   it('rejects double spend during reorg', async () => {
```

Comment:
After some thinking, I think we should delete this test too. This is testing the same thing as the above test, that we check for double spends inside the pre-connect when were connecting a block.

At most, if you wanted to add a test that demonstrates you can add a double spend to a fork that would be different but that's expected behaviour.

Reply:
Idk I think this is still useful. It is a test that demonstrates you can add a double spend to a fork, like you said, and also tests that `connect` is called correctly when doing a re-org. Or at least that the check for double spends is actually IN the connect function and not just in `addBlock`

Reply:
That's fine to keep it if you want.

You could demonstrate that without adding a double spend, though. Add the TX to the main chain, and the fork, and it'll show the spends aren't validated on the fork because it would have caught it from the main chain block with that same TX.